File: Leetcode/inorderTraversal.swift
func inorderTraversal(root: TreeNode?) -> [Int] {
        var result : [Int] = []
        var stack: [TreeNode] = []
        pushLeft(root, &stack)
        while !stack.isEmpty {
            var current = stack.popLast()!
            result.append(current.val)
            pushLeft(current.right, &stack)
        }
        return result
    }
    func pushLeft(root: TreeNode?, inout _ stack: [TreeNode]) {
        var current = root
        while let node = current {
            stack.append(node)
            current = current?.left
        }
    }

File: Leetcode/inorderTraversal.java
public List<Integer> inorderTraversal(TreeNode root) {
        ArrayList<Integer> result = new ArrayList<Integer>();
        if (root == null) {
            return result;
        }
        Stack<TreeNode> stack = new Stack<TreeNode>();
        pushLeft(root, stack);
        while (!stack.empty()) {
            TreeNode current = stack.pop();
            result.add(current.val);
            pushLeft(current.right, stack);
        }
        return result;
    }
    private void pushLeft(TreeNode root, Stack<TreeNode> stack) {
        TreeNode p = root;
        while (p != null) {
            stack.push(p);
            p = p.left;
        }
    }
